Avoid using non-UTF-8 encoded argv more often

dnfield opened the pull request:
Should fix https://github.com/flutter/flutter/issues/112819

I have not tested this on a windows machine though. @sealesj would you be able to kick off a LED run with this patched in for one of your failing windows tests?

dnfield commented on fml/command_line.h:
```diff
@@ -223,7 +223,14 @@ inline CommandLine CommandLineFromIteratorsWithArgv0(const std::string& argv0,
 // Builds a |CommandLine| from the usual argc/argv.
+//
+// First checks if CommandLineFromPlatform returns a value, to avoid using an
+// argv that is not provided as UTF-8.
```
tester_main.cc uses this method, and now will pick up the UTF-8 encoded args on windows.

chinmaygarde replied:
I think this will break embedders who provide argc and argv explicitly.

chinmaygarde replied:
Here https://github.com/flutter/engine/blob/e6e94e4ce65290d2c4db7403689a80bc11c2868f/shell/platform/embedder/embedder.cc#L1331

dnfield replied:
It would mean we ignore the specified argc/argv on Windows and instead use the platform based method. I'm not really clear on how that's used today, but I suspect that embedders  on Windows doing this are currently broken unless they're already using this API.

I suppose this could be a problem if the embedder is specifying values for argc/argv that weren't already passed in to the process though...

dnfield replied:
I'll refactor this to a separate method though to avoid disrupting the embedder API.

dnfield commented:
The test for this is in the linked bug.

jason-simmons commented:
The declaration for `CommandLineFromPlatform` must be moved above `CommandLineFromArgcArgv`

Otherwise LGTM

Verified that `flutter test` passes with this patch when run on Windows from a non-ASCII directory path.

sealesj commented:
Kicked off and tried to patch in both the gerrit utf-8 and this PR. I think I did it correctly, but still seeing errors https://luci-milo.appspot.com/raw/build/logs.chromium.org/flutter/led/jseales_google.com/a7913409ba5ef249a24e399ca49b9a15f8def9e09a3115737d86763096077309/+/build.proto

dnfield commented:
@jason-simmons - I rearranged the methods

@chinmaygarde - I added a new method and updated all the callsites except embedder.cc and one unit test explicitly testing the old method.

dnfield commented:
Ahh that task is trying to pull from the framework instead of the engine, so it won't work.
